doc/services/mem_mgmt/index.rst

```diff
@@ -84,7 +84,101 @@ one by renaming the property and changing its value according to the following l
    "IO"          -> <( DT_ARM_MPU(ATTR_MPU_IO) )>
    "EXTMEM"      -> <( DT_ARM_MPU(ATTR_MPU_EXTMEM) )>
 
+Memory Attributes Heap Allocator
+********************************
+
+It is possible to leverage the memory attribute property ``zephyr,memory-attr``
+to define and create a set of memory heaps from which the user can allocate
+memory from with certain attributes / capabilities.
+
+When the :kconfig:option:`CONFIG_MEM_ATTR_HEAP` is set, every region marked
+with one of the memory attributes listed in in
+:zephyr_file:`include/zephyr/dt-bindings/memory-attr/memory-attr-sw.h` is added
+to a pool of memory heaps used for dynamic allocation of memory buffers with
+certain attributes.
+
+Here a non exhaustive list of possible attributes:
+
+.. code-block:: none
+
+   DT_MEM_SW_ALLOC_CACHE
+   DT_MEM_SW_ALLOC_NON_CACHE
+   DT_MEM_SW_ALLOC_DMA
```

Comment:
It worth to mention here if it is possible to specify multiple `DT_MEM_SW_ALLOC*` attributes. Like `DT_MEM_SW_ALLOC_DMA | DT_MEM_SW_ALLOC_NON_CACHE`.

As I understand that wouldn't work (please correct me if I'm wrong) - so we should mention that in the docs and it would be useful to add compile-time assertion against such combination. 

Alternatively we can just define `DT_MEM_SW_ALLOC*` attributes not as a individual bits but as an enum:
```diff
-#define  ATTR_SW_ALLOC_CACHE           BIT(0)
-#define  ATTR_SW_ALLOC_NON_CACHE       BIT(1)
-#define  ATTR_SW_ALLOC_DMA             BIT(2)
+#define  ATTR_SW_ALLOC_CACHE           1
+#define  ATTR_SW_ALLOC_NON_CACHE       2
+#define  ATTR_SW_ALLOC_DMA             3
```

Reply:
>As I understand that wouldn't work (please correct me if I'm wrong) 

It would work. From the allocator prospective `DT_MEM_SW_ALLOC_CACHE` is different than `DT_MEM_SW_ALLOC_CACHE | DT_MEM_SW_ALLOC_DMA`, so you can allocate from one region or from the other region depending on the parameter passed to `mem_attr_heap_alloc()`.

I'll add a new test for this case and I'll fix the documentation that indeed is a bit unclear on this point.